040-SDK/160-image-transformations.mdx

```diff
@@ -52,22 +83,32 @@ https://us-east-1.storage.xata.sh/transform/height=50,format=webp/4u1fh2o6p10blb
 
 </TabbedCode>
 
-With an existing Xata image URL you can also perform a similar transformation using the `transformImage` function imported from the TypeScript client. Performing a transform client-side can be useful for generating images conditionally based on media queries within a browser. With this method, only a `url` for an existing Xata image is needed.
+## Post-transformation metadata
 
-```ts title="client_component.tsx"
-'use client';
-import { transformImage } from '@xata.io/client';
+Once a transformation occurs, it is common to need the dimensions of the new image. `metaDataUrl` and `signedMetaUrl` (if using signed URLs) are available when making transformations from a file on a record.
 
-// Once you have a URL to a Xata image, you can transform it and return a new URL
-const transformedImageUrl = transformImage('https://us-east-1.storage.xata.sh/4u1fh2o6p10blbutjnphcste94', {
+```ts
+// Get a specific record from the "gallery" database
+const record = await xata.db.gallery.read('record_id');
+
+// Also grab the related meta data (height, width...etc) of the new image
+const { url, metaDataUrl } = record.imageColumnName.transform({
   height: 100,
   rotate: 180,
   format: 'webp'
 });
+
+console.log('height of new image', metaDataUrl.height); // 100px
```

Comment:
Nope, you still need to fetch the result. It's a URL not the response of the URL